kernel/src/schema/diff.rs

```diff
@@ -0,0 +1,327 @@
+//! Schema diffing implementation for Delta Lake schemas
+//!
+//! This module provides functionality to compute differences between two schemas
+//! using field IDs as the primary mechanism for identifying fields across schema versions.
+//! Supports nested field comparison within structs, arrays, and maps.
+
+// Allow dead code warnings since this API is not yet used by other modules
+#![allow(dead_code)]
+// TEMPORARY: Allow unused imports in PR 1 - these will be used when the full implementation is added in PR 2
+#![allow(unused_imports)]
+
+use super::{ColumnMetadataKey, ColumnName, DataType, MetadataValue, StructField, StructType};
+use std::collections::{HashMap, HashSet};
+
+/// Feature gate for schema diff functionality.
+/// Set to `false` to ensure incomplete implementations don't activate until all PRs are merged.
+/// Will be removed in the final PR when all tests and implementation are complete.
+const SCHEMA_DIFF_ENABLED: bool = false;
+
+/// Arguments for computing a schema diff
+#[derive(Debug, Clone)]
+pub(crate) struct SchemaDiffArgs<'a> {
+    /// The before/original schema
+    pub before: &'a StructType,
```

Comment:
not sure how others feel but generally for `pub(crate)` fields i like the clarity of making each one `pub(crate)` instead of `pub` (which is the same as `pub(crate) as long as the struct itself is `pub(crate)`)